File: csharp/ASCrypt/MD5.cs
using System;
using System.Security.Cryptography;

namespace ASCrypt
{
    public class MD5
    {
        /// <summary>
        /// Computes the MD5 checksum for the bytes.
        /// </summary>
        public static Byte[] Compute(Byte[] bytes)
        {
            MD5CryptoServiceProvider md5 = new MD5CryptoServiceProvider();
            return md5.ComputeHash(bytes);
        }

        /// <summary>
        /// Computes the HMAC-MD5 for the key and bytes.
        /// </summary>
        public static Byte[] ComputeHMAC(Byte[] key, Byte[] bytes)
        {
            HMACMD5 hmac = new HMACMD5(key);
            return hmac.ComputeHash(bytes);
        }

    }

}

File: php/classes/MD5.php
<?php

class MD5
{
	/**
	* Computes the MD5 checksum.
	*/	
	public static function compute($text)
	{
		return hash("md5", $text, true);
	}
	
	/**
	* Computes the HMAC for MD5.
	*/
	public static function computeHMAC($key, $text)
	{
		return hash_hmac("md5", $text, $key, true);
	}
	
}

?>
